### PR TITLE
Update external dependency retrieval from WORKSPACE to MODULE to support bzlmod

### PR DESCRIPTION
**Pull request by phaedon**
This is a first step to support the changes outlined in https://blog.bazel.build/2023/07/24/whats-new-with-bzlmod.html

### CODE REVIEW

File: MODULE.bazel [hunk 1]
```diff
@@ -0,0 +1,11 @@
+bazel_dep(name = "abseil-cpp", version = "20230802.0", repo_name = "com_google_absl")
+bazel_dep(name = "abseil-py", version = "1.4.0", repo_name = "com_google_absl_py")
+bazel_dep(name = "bazel_skylib", version = "1.5.0")
+bazel_dep(name = "grpc", version = "1.48.1")
```
Comment (laramiel) [inline MODULE.bazel:4]: This change removes the usage dependency on grpc proto rules.

Comment (phaedon) [reply MODULE.bazel]: Good catch, thanks!

**Comment by rwgk** [comment]
LGTM — But I don't have a lot of relevant background.

@laramiel could you please confirm?

I'll need to import this change manually. (The source of truth for this repo is the Google-internal SCM system.) This change will then get auto-exported from there.

**Comment by phaedon** [comment]
Is this PR making its way through the system? Once it's there, we should consider adding a versioned release so that we can add this library to the BCR (https://registry.bazel.build/)

**Comment by rwgk** [comment]
Just to make a connection that may be relevant (not sure): pybind/pybind11_bazel#72

@junyer for awareness.

**Comment by junyer** [comment]
Supporting Bzlmod is on @david-crouse's radar. :D

**Comment by phaedon** [comment]
This is resolved thanks to https://github.com/pybind/pybind11_protobuf/commit/b4a2e87a10cd5f6309e4ff67c040a470d7ec2373

**Pull request closed without merging**